src/caffe/layers/batch_reindex_layer.cpp

```diff
@@ -10,6 +10,7 @@ void BatchReindexLayer<Ftype, Btype>::Reshape(const vector<Blob*>& bottom,
                                        const vector<Blob*>& top) {
   CHECK_EQ(1, bottom[1]->num_axes());
   vector<int> newshape;
+  newshape.reserve(bottom[0]->shape().size() + 1);
```

Comment:
May be better
```
const auto& shp0 = bottom[0]->shape();
vector<int> newshape(shp0.size());
newshape[0]=bottom[1]->shape()[0];
for (int i = 1; i < shp0.size(); ++i) {
    newshape[i]=shp0[i];
}
```
? Also, size+1 allocates 1 redundant entry.

Reply:
You are right making the temp is better! 

the size+1 because notice in addition to the loop(that pushed size elements), before the loop it pushes one more at
```newshape.push_back(bottom[1]->shape(0));```` 

Here notice: 
```
 newshape.push_back(bottom[1]->shape(0));  <-------------------[1-pushed]
 for (int i = 1; i < bottom[0]->shape().size(); ++i) {
    newshape.push_back(bottom[0]->shape()[i]); <-----------------[....size()-pushed]
  }
// So total [size + 1] pushed.
```

Reply:
Doing the constructor way does default initialization(size times) and reserve(size capacity), but then the initialization is unnecessary because we just re-assign over it? Take a look at what i did, this way the push logic still stays, and you don't initialize every thing initially to a default constructed state.